From 08ff0fab6781a10e8cdf82d097bbedd947ae6fc9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Wed, 20 May 2020 14:15:34 +0200
Subject: [PATCH 1/1] Utils: Add manament command to load stats

This is sometimes good thing to perform after upgrade to avoid
uwsgi workers racing for the stats calculation.
---
 .../utils/management/commands/ensure_stats.py | 28 +++++++++++++++++++
 1 file changed, 28 insertions(+)
 create mode 100644 weblate/utils/management/commands/ensure_stats.py

diff --git weblate/utils/management/commands/ensure_stats.py weblate/utils/management/commands/ensure_stats.py
new file mode 100644
index 0000000000..80c96199e9
--- /dev/null
+++ b/weblate/utils/management/commands/ensure_stats.py
@@ -0,0 +1,28 @@
+#
+# Copyright © 2012 - 2020 Michal Čihař <michal@cihar.com>
+#
+# This file is part of Weblate <https://weblate.org/>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from weblate.utils.management.base import BaseCommand
+from weblate.utils.stats import GlobalStats
+
+
+class Command(BaseCommand):
+    help = "ensures that stats are present"
+
+    def handle(self, *args, **options):
+        GlobalStats().ensure_basic()
-- 
2.27.0.rc0

